src/client/pfs/pfs.proto

```diff
@@ -94,9 +94,19 @@ message CommitRange {
   Commit upper = 2;
 }
 
+// CommitOrigin keeps track of where (i.e. which branch) a certain commit
+// originated from. A commits provenance consists of the commits of
+// the commits which are in its causal history.
+message CommitOrigin {
+  Commit commit = 1;
+  Branch branch = 2;
+}
+
 // CommitInfo is the main data structure representing a commit in etcd
 message CommitInfo {
+  reserved 6, 10;
   Commit commit = 1;
+  Branch original_branch = 15;
```

Comment:
Let's change this to `branch`, similar to the logic below, while `original_branch` is a little more descriptive I think it's only in very specific situations that there will be multiple branches so I prefer the terseness. This would also be a good place to document that this is the branch the commit was created on.